src/sources/opentelemetry/log.rs

```diff
@@ -0,0 +1,254 @@
+use crate::{
+    config::{
+        AcknowledgementsConfig, DataType, GenerateConfig, Output, Resource, SourceConfig,
+        SourceContext,
+    },
+    internal_events::{EventsReceived, StreamClosedError},
+    opentelemetry::LogService::{
+        logs_service_server::{LogsService, LogsServiceServer},
+        ExportLogsServiceRequest, ExportLogsServiceResponse,
+    },
+    serde::bool_or_struct,
+    sources::{util::grpc::run_grpc_server, Source},
+    tls::{MaybeTlsSettings, TlsEnableableConfig},
+    SourceSender,
+};
+use futures::TryFutureExt;
+use serde::{Deserialize, Serialize};
+use std::net::SocketAddr;
+use tonic::{Request, Response, Status};
+use vector_core::{
+    config::LogNamespace,
+    event::{BatchNotifier, BatchStatus, BatchStatusReceiver, Event},
+    ByteSizeOf,
+};
+
+pub const LOGS: &str = "logs";
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct OpentelemetryConfig {
```

Comment:
We'll probably want to pull up these bits into `mod.rs` but I think this is fine until we add support for more otel data types.

Reply:
That was on my list to refactor once we merged 👍